tests/Basics/InifileParserTest.cpp

```diff
@@ -0,0 +1,161 @@
+
+////////////////////////////////////////////////////////////////////////////////
+/// @brief test_parsing
+////////////////////////////////////////////////////////////////////////////////
+
+SECTION("test_parsing") {
+  using namespace arangodb::options;
+
+  uint64_t writeBufferSize = UINT64_MAX;
+  uint64_t totalWriteBufferSize = UINT64_MAX;
+  uint64_t maxWriteBufferNumber = UINT64_MAX;
+  uint64_t maxTotalWalSize = UINT64_MAX;
+  uint64_t blockCacheSize = UINT64_MAX;
+  bool enforceBlockCacheSizeLimit = false;
+  uint64_t cacheSize = UINT64_MAX;
+  uint64_t nonoSetOption = UINT64_MAX;
+  uint64_t someValueUsingSuffixes = UINT64_MAX;
+  uint64_t someOtherValueUsingSuffixes = UINT64_MAX;
+  uint64_t yetSomeOtherValueUsingSuffixes = UINT64_MAX;
+  uint64_t andAnotherValueUsingSuffixes = UINT64_MAX;
+  uint64_t andFinallySomeGb = UINT64_MAX;
+  
+  ProgramOptions options("testi", "testi [options]", "bla", "/tmp/bla");
+  options.addSection("rocksdb", "bla");
+  options.addOption("--rocksdb.write-buffer-size", "bla", new UInt64Parameter(&writeBufferSize));
+  options.addOption("--rocksdb.total-write-buffer-size", "bla", new UInt64Parameter(&totalWriteBufferSize));
+  options.addOption("--rocksdb.max-write-buffer-number", "bla", new UInt64Parameter(&maxWriteBufferNumber));
+  options.addOption("--rocksdb.max-total-wal-size", "bla", new UInt64Parameter(&maxTotalWalSize));
+  options.addOption("--rocksdb.block-cache-size", "bla", new UInt64Parameter(&blockCacheSize));
+  options.addOption("--rocksdb.enforce-block-cache-size-limit", "bla", new BooleanParameter(&enforceBlockCacheSizeLimit));
+  
+  options.addSection("cache", "bla");
+  options.addOption("--cache.size", "bla", new UInt64Parameter(&cacheSize));
+  options.addOption("--cache.nono-set-option", "bla", new UInt64Parameter(&nonoSetOption));
+
+  options.addSection("pork", "bla");
+  options.addOption("--pork.some-value-using-suffixes", "bla", new UInt64Parameter(&someValueUsingSuffixes));
+  options.addOption("--pork.some-other-value-using-suffixes", "bla", new UInt64Parameter(&someOtherValueUsingSuffixes));
+  options.addOption("--pork.yet-some-other-value-using-suffixes", "bla", new UInt64Parameter(&yetSomeOtherValueUsingSuffixes));
+  options.addOption("--pork.and-another-value-using-suffixes", "bla", new UInt64Parameter(&andAnotherValueUsingSuffixes));
+  options.addOption("--pork.and-finally-some-gb", "bla", new UInt64Parameter(&andFinallySomeGb));
+
+  auto contents = R"data(
+[rocksdb]
+# Write buffers
+write-buffer-size = 2048000 # 2M
+total-write-buffer-size = 536870912
+max-write-buffer-number = 4
+max-total-wal-size = 1024000 # 1M
+
+# Read buffers
+block-cache-size = 268435456
+enforce-block-cache-size-limit = true
+
+[cache]
+size = 268435456 # 256M
+
+[pork]
+some-value-using-suffixes = 1M
+some-other-value-using-suffixes = 1MiB
+yet-some-other-value-using-suffixes = 12MB  
+   and-another-value-using-suffixes = 256kb  
+   and-finally-some-gb = 256GB
+)data";
```

Comment:
I suggest to also add one line/check where a `#` occurs in the value, so we secure that behaviour as well.

Reply:
I was a little inarticulate, sorry. What I meant was, adding some option that is not numeric, i.e. a `StringParameter`, where the `#` should not be removed.